### DISCUSSION TITLE
export type PostgrestFilterBuilder for typescript

### DISCUSSION THREAD

**Discussion by d-mok** [supabase/supabase#2733] [Feature Requests]
Can we export the type `PostgrestFilterBuilder` for typescript, so that we can write functions which dynamically build the filter part of a select query? Now, when I write something like:
```
function addOrFilter(query: PostgrestFilterBuilder, filter:string){
   return query.or(filter)
}
// query is something like supabase.from('table').select('*')
```
I encounter the error:
```
Module '"@supabase/supabase-js"' has no exported member 'PostgrestFilterBuilder'.
```

**Comment by dwome** [comment]
I am also looking forward to this type. Is this somehow already possible? ☺️

**Reply by dwome** [reply]
You can use:
import type { PostgrestFilterBuilder } from "@supabase/postgrest-js";

**Comment by hmnd** [comment]
Although it is already possible to import this type from `@supabase/postgrest-js`, `@supabase/supabase-js` maintains its own dependency on `@supabase/postgrest-js`, making it easy for the types in both packages to fall out of sync with each other due to conflicting versions. I would prefer if either `PostgrestFilterBuilder` was re-exported from `supabase-js`, or if `supabase-js` listed `postgrest-js` as a peer dep instead.

**Reply by wanapichpongsa** [reply]
Why has this still not been done yet???

**Comment by cvburgess** [comment]
This would be really helpful! I'm going to try the solution above but worry about version mismatches
